nvflare/tool/code_pre_installer/README.md

```diff
@@ -0,0 +1,255 @@
+# NVFLARE Code Pre-Installer
+
+This tool helps install NVFLARE application code and libraries before running federated learning jobs.
+
+## Overview
+
+The code pre-installer handles:
+- Installation of application code
+- Installation of shared libraries
+- Site-specific customizations
+- Python package dependencies
+
+## Directory Structure
+
+Expected application code zip structure:
+
+```
+application.zip
+├── application/<app_name>/
+│               ├── meta.json       # Application metadata
+│               ├── app_<site>/     # Site custom code
+│                  └── custom/      # Site custom code
+├── application-share/              # Shared resources
+│   └── shared.py
+└── requirements.txt       # Python dependencies (optional)
+```
+or
+```
+application.zip
+├── application/<app_name>/
+│               ├── meta.json       # Application metadata
+│               ├── app/            # Site custom code
+│                  └── custom/      # Site custom code
+``` 
+Here is an example of creating a folder structure for pre-installation:
+```bash
+mkdir -p /tmp/nvflare/pre-install/application
+mkdir -p /tmp/nvflare/pre-install/application-share
+```
+
+For example, if the app name is `fedavg`, the directory structure would look like this:
+
+Tree structure of the job configuration:
+
+/tmp/nvflare/pre-install/
+├── application
+│   └── fedavg
+│       ├── app_server
+│       │   ├── config
+│       │   └── custom
+│       ├── app_site-1
+│       │   ├── config
+│       │   └── custom
+│       ├── app_site-2
+│       │   ├── config
+│       │   └── custom
+│       ├── app_site-3
+│       │   ├── config
+│       │   └── custom
+│       ├── app_site-4
+│       │   ├── config
+│       │   └── custom
+│       ├── app_site-5
+│       │   ├── config
+│       │   └── custom
+│       └── meta.json
+└── application-share
+    └── pt
+        ├── learner_with_mlflow.py
+        ├── learner_with_tb.py
+        ├── learner_with_wandb.py
+        ├── pt_constants.py
+        ├── simple_network.py
+        └── test_custom.py
```

Comment:
should put inside "```" for better formatting

Reply:
When I open https://github.com/NVIDIA/NVFlare/blob/b9ba0092b06d5451e9ca2ad9944228024fdb3fc2/nvflare/tool/code_pre_installer/README.md
the rendering is OFF in my browser.
other part inside "```" is OK